### PR TITLE
Merge release branch to master

### PR DESCRIPTION
**Pull request by tobrun**
This PR merges 'release-ios-v3.5.0-android-v5.0.0' back to master.

cc @mapbox/gl @mapbox/mobile

### CODE REVIEW

File: platform/ios/CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,13 @@
...
-## master
+## 3.5.0
+
+### Packaging
+
+* The minimum deployment target for this SDK is now iOS 8. ([#8129](https://github.com/mapbox/mapbox-gl-native/pull/8129))
+* Added support for the Carthage dependency manager. See [our website](https://www.mapbox.com/ios-sdk/) for setup instructions. ([#8257](https://github.com/mapbox/mapbox-gl-native/pull/8257))
+* While running your application in the iOS Simulator, you will receive a notice in the console if a newer version of this SDK is available. ([#8282](https://github.com/mapbox/mapbox-gl-native/pull/8282))
 
 * The error passed into `-[MGLMapViewDelegate mapViewDidFailLoadingMap:withError:]` now includes a more specific description and failure reason. ([#8418](https://github.com/mapbox/mapbox-gl-native/pull/8418))
```
Comment (1ec5) [inline platform/ios/CHANGELOG.md]: This note needs to go in a new master section at the top.

File: platform/macos/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Changelog for Mapbox macOS SDK
 
-## master
+## 0.4.0
 
 * The error passed into `-[MGLMapViewDelegate mapViewDidFailLoadingMap:withError:]` now includes a more specific description and failure reason. ([#8418](https://github.com/mapbox/mapbox-gl-native/pull/8418))
```
Comment (1ec5) [inline platform/macos/CHANGELOG.md]: Likewise, this note needs to go in a new master section up top.

**Comment by 1ec5** [comment]
With b549b0e1453ffba2c178565f3359fe5e9ca35f3c and the change I've requested, you may want to redo the merge, just to be safe.

**Comment by jfirebaugh** [comment]
I rebased the merge, squashing b549b0e1453ffba2c178565f3359fe5e9ca35f3c, the changes @1ec5 requested, and a couple other fixes. @1ec5 can you re-review?

**Comment by jfirebaugh** [comment]
I don't think I can do that easily.

**Pull request merged**